lib/motion/channel.rb

```diff
@@ -65,15 +65,15 @@ def process_periodic_timer(timer)
     private
 
     def synchronize
+      component_connection.if_render_required do |component|
+        transmit(renderer.render(component))
+      end
+
```

Comment:
Putting the render *before* the stream synchronization allows new streams to be setup during render. Without this, `bind` would not work in the template.

Even without the `bind` feature, this probably is the better ordering since it moves all of the channel synchronization to after the user code. I just never noticed before because it is weird for templates to have side-effects.